cmd/system-shutdown/system-shutdown-utils.c

```diff
@@ -66,6 +66,27 @@ void die(const char *msg)
 	exit(1);
 }
 
+int read_reboot_arg(char *arg, int max_size)
+{
+	FILE *f;
+
+	*arg = '\0';
+
+	f = fopen("/run/systemd/reboot-param", "r");
```

Comment:
Nope, it is the same way used by systemd:

https://github.com/systemd/systemd/blob/master/src/systemctl/systemctl.c#L8262